client/src/components/DynamicLink.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCanEquip, useMyHash, useToInt } from "../hooks/useCall";
+import { $item, Placeholder } from "../util/makeValue";
+import MainLink from "./MainLink";
+
+interface Props {
+  linkedContent:
+    | Placeholder<"Item">
+    | Placeholder<"Familiar">
+    | Placeholder<"Skill">;
+}
+
+const DynamicLink: React.FC<Props> = ({ linkedContent }) => {
+  const myHash = useMyHash() ?? 0;
+  const isItem = linkedContent.objectType === "Item";
+  const isFamiliar = linkedContent.objectType === "Familiar";
+  const isSkill = linkedContent.objectType === "Skill";
+  const linkID = useToInt(linkedContent) ?? 1;
+  const isEquippable =
+    +isItem * +(useCanEquip($item`${linkID + ""}`) ?? 0) === 1;
+
+  if (isItem) {
```

Comment:
i would turn this into a switch statement on `linkedContent.objectType` and get rid of the isXXXs. Type system will verify what the possible values are so you won't even have to have the dead empty return at the end.